internal/armada/repository/job.go

```diff
@@ -56,7 +56,7 @@ type JobResult struct {
 
 type JobRepository interface {
 	PeekQueue(queue string, limit int64) ([]*api.Job, error)
-	TryLeaseJobs(clusterId string, queue string, jobs []*api.Job) ([]*api.Job, error)
+	TryLeaseJobs(clusterId string, jobIdsByQueue map[string][]string) (map[string][]string, error)
```

Comment:
can we document what's going on here?  Specifically what does the returned `map[string][]string` represent

Reply:
Done